server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -80,21 +81,33 @@ public void done(boolean success) {
             .filter(getResponse -> getResponse.getBody() != null);
     }
 
-    Flux<MailQueue.MailQueueItem> deQueue() {
-        return flux.flatMap(this::loadItem);
+    Flux<? extends MailQueue.MailQueueItem> deQueue() {
+        return flux.flatMap(this::loadItem)
+            .concatMap(this::filterIfDeleted);
```

Comment:
I do not see the difference with the last attempt which broke up some tests.

Reply:
Yes this fix can [should] be discarded